### PR TITLE
Include squad analysis into email results report

### PR DESCRIPTION
**Pull request by dahorak**
Controlled by `--squad-analysis` parameter.

Fixes https://github.com/red-hat-storage/ocs-ci/issues/2920

Signed-off-by: Daniel Horak <dahorak@redhat.com>

### CODE REVIEW

**Review by ebenahar — APPROVED** [review]

File: ocs_ci/framework/pytest_customization/reports.py [hunk 1]
```diff
@@ -43,11 +43,21 @@ def pytest_runtest_makereport(item, call):
...
+    session.results = dict()
 
 
 def pytest_sessionfinish(session, exitstatus):
     """
     send email report
     """
     if ocsci_config.RUN['cli_params'].get('email'):
-        email_reports()
+        email_reports(session)
```
Comment (vasukulkarni) [inline ocs_ci/framework/pytest_customization/reports.py:63]: how does this work with multiple pipelines?

**Review by clacroix12 — APPROVED** [review]
LGTM!

File: ocs_ci/framework/pytest_customization/reports.py (hunk 1 above)
Comment (dahorak) [reply ocs_ci/framework/pytest_customization/reports.py:63]: @vasukulkarni what do you men by _multiple pipelines_?

File: ocs_ci/ocs/constants.py [hunk 2]
```diff
@@ -1142,3 +1142,20 @@
 MCG_NS_RESOURCE = 'ns_resource'
 MCG_NS_BUCKET = 'ns-bucket'
 MCG_NS_AWS_CONNECTION = 'aws_connection'
+
+# Create your own squads.
+# Squads are separated by Python packages in your project so it is important to
+# put the squad name in '.' --> ".squad_name." For example: In case this test
+# fails - tests.e2e.registry.test_pod_from_registry.TestRegistryImage.test_run_pod_local_image
+# the squad is ".registry." as can be seen in the Magenta squad below
```
Comment (vavuthu) [inline ocs_ci/ocs/constants.py:1150]: below SQUADS dict is not using ".squad_name" or ".registry". Can we remove or modify the comment accordingly

File: ocs_ci/utility/utils.py [hunk 3]
```diff
@@ -934,7 +934,119 @@ def parse_html_for_email(soup):
     main_header.string.replace_with('OCS-CI RESULTS')
 
 
-def email_reports():
+def add_squad_analysis_to_email(session, soup):
+    """
+
```
Comment (vavuthu) [inline ocs_ci/utility/utils.py:939]: Could you add description

Comment (dahorak) [reply ocs_ci/utility/utils.py:940]: Fixed by 8c87c59.

File: ocs_ci/ocs/constants.py (hunk 2 above)
Comment (dahorak) [reply ocs_ci/ocs/constants.py]: Good point! Fixed by 8c87c59.

**Review by vavuthu — DISMISSED** [review]

File: ocs_ci/ocs/constants.py [hunk 4]
```diff
@@ -1142,3 +1142,21 @@
...
+# Tests are assigned to Squads based on patterns matching test path.
+# For example: In case following test fails:
+# tests/e2e/registry/test_pod_from_registry.py::TestRegistryImage::test_run_pod_local_image
+# the pattern "/registry/" match the test path and so the test belongs to
+# Magenta squad.
+SQUADS = {
+    'Brown': ["/nodes/"],
+    'Green': ["/pv_services/", "/storageclass/"],
+    'Blue': ["/monitoring/"],
+    'Red': ["/mcg/"],
```
Comment (MeridianExplorer) [inline ocs_ci/ocs/constants.py:1156]: Also needs `/rgw/` :)

Comment (MeridianExplorer) [inline ocs_ci/ocs/constants.py:1156]:
```suggestion
    'Red': ["/mcg/", "/rgw/"],
```
The red team is also responsible for the RGW tests

Comment (dahorak) [reply ocs_ci/ocs/constants.py]: Added dc2a4fb.

File: ocs_ci/utility/utils.py [hunk 5]
```diff
@@ -934,7 +934,120 @@ def parse_html_for_email(soup):
...
 
 
-def email_reports():
+def add_squad_analysis_to_email(session, soup):
+    """
+    Add squad analysis to the html test results used in email reporting
+
+    Args:
+        session (obj): Pytest session object
+        soup (obj): BeautifulSoup object
```
Comment (petr-balogh) [inline ocs_ci/utility/utils.py:943]: BeautifulSoap object of what?  Maybe worth to add like of HTML report data?

File: ocs_ci/framework/pytest_customization/ocscilib.py [hunk 6]
```diff
@@ -84,6 +84,13 @@ def pytest_addoption(parser):
         dest='email',
         help="Email ID to send results",
     )
+    parser.addoption(
+        '--squad-analysis',
```
Comment (petr-balogh) [inline ocs_ci/framework/pytest_customization/ocscilib.py:88]: @dahorak please add some info about this parameter also to documentation: https://github.com/red-hat-storage/ocs-ci/blob/master/docs/usage.md

**Review by petr-balogh — COMMENTED** [review]
Otherwise looks good to me

File: ocs_ci/utility/utils.py (hunk 5 above)
Comment (dahorak) [reply ocs_ci/utility/utils.py]: Copy&paste from other method, fixed in 21b7e01.

File: ocs_ci/framework/pytest_customization/ocscilib.py (hunk 6 above)
Comment (dahorak) [reply ocs_ci/framework/pytest_customization/ocscilib.py:88]: Added in 21b7e01.

**Review by petr-balogh — APPROVED** [review]

**Comment by dahorak** [comment]
The last commit eb37112 contains following changes:
* do not change email report if there are any failed or skipped test (do not add empty Squad Analysis ... section)
* Consider also tests which are not assigned to any squad and print them in the result (with highlighted background):
email-results-squad-analysis-example2

Will wait for PR validation job, to get actual result from tier1 tests.

**Review by vavuthu — APPROVED** [review]

**Review by vasukulkarni — APPROVED** [review]

**Comment by dahorak** [comment]
Real email from `tier1` tests execution (the last PR validation job).
email-results-squad-analysis-tier1

**Pull request merged**